src/overlays/actors/ovl_En_Elf/z_en_elf.c

```diff
@@ -21,99 +66,1510 @@ const ActorInit En_Elf_InitVars = {
+            }
+        } else {
+            if (xzDistFromLink > 150.0f) {
+                xzDistFromLink = 150.0f;
+            }
+
+            xzDistFromLink = ((xzDistFromLink - 50.0f) * 0.95f) + 0.05f;
+
+            if (Math_Rand_ZeroOne() < xzDistFromLink) {
+                this->unk_2A8 = 3;
+                this->unk_2AC = 0x200;
+                this->unk_2B8 = (xzDistFromLink + xzDistFromLink) + 1.0f;
+                this->func_2C8 = func_80A01FE0;
+                this->unk_2C0 = (s16)Math_Rand_ZeroFloat(16.0f) + 0x10;
+            } else {
+                this->unk_2C0 = 10;
+            }
+        }
+    }
+
+    if (Math_Rand_ZeroOne() < 0.1f) {
+        this->unk_2A8 = 1;
+        this->unk_2AC = 0x80;
+        this->unk_2B8 = Math_Rand_ZeroFloat(0.5f) + 0.5f;
+        this->unk_2B0 = Math_Rand_CenteredFloat(32767.0f);
+    }
+}
+
+void func_80A0232C(EnElf* this, GlobalContext* globalCtx) {
+    if (func_80A01F90(&this->unk_28C, &this->actor.posRot.pos, 100.0f)) {
+        this->unk_2A8 = 0;
+        this->unk_2AC = 0x200;
+        this->func_2C8 = func_80A0214C;
+        this->unk_2B8 = 1.5f;
+    } else {
+        this->func_2C8(this, globalCtx);
+    }
+}
+
+f32 EnElf_GetColorValue(s32 colorFlag) {
+    switch (colorFlag) {
+        case 1:
+            return Math_Rand_ZeroFloat(55.0f) + 200.0f;
+        case 2:
+            return Math_Rand_ZeroFloat(255.0f);
+        default:
+            return 0.0f;
+    }
+}
+
+void EnElf_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnElf* this = THIS;
+    s32 pad;
+    Player* player = PLAYER;
+    s32 colorConfig;
+    s32 i;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_04016A48, &D_04014BA4, this->limbDrawTable,
+                   this->transitionDrawTable, 15);
+    ActorShape_Init(&thisx->shape, 0.0f, NULL, 15.0f);
+    thisx->shape.unk_14 = 0xFF;
+
+    Lights_PointGlowSetInfo(&this->lightInfoGlow, thisx->posRot.pos.x, thisx->posRot.pos.y, thisx->posRot.pos.z, 255,
+                            255, 255, 0);
+    this->lightNodeGlow = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfoGlow);
+
+    Lights_PointNoGlowSetInfo(&this->lightInfoNoGlow, thisx->posRot.pos.x, thisx->posRot.pos.y, thisx->posRot.pos.z,
+                              255, 255, 255, 0);
+    this->lightNodeNoGlow = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfoNoGlow);
+
+    this->flags = 0;
+    this->dissapearTimer = 600;
+    this->unk_2A4 = 0.0f;
+    colorConfig = 0;
+
+    switch (thisx->params) {
+        case FAIRY_NAVI:
+            thisx->room = -1;
+            EnElf_SetupAction(this, func_80A03CF8);
+            func_80A01C38(this, 0);
+            this->flags |= 4;
+            thisx->update = func_80A053F0;
+            this->elfMsg = NULL;
+            this->unk_2C7 = 0x14;
+
+            if ((gSaveContext.naviTimer >= 25800) || (gSaveContext.naviTimer < 3000)) {
```

Comment:
```suggestion
            if (gSaveContext.naviTimer >= 25800 || gSaveContext.naviTimer < 3000) {
```
death to all parens Pepega